infobip-typescript-generator-extension-api/src/main/java/com/infobip/typescript/transformer/ClassTransformerDecoratorExtension.java

```diff
@@ -114,6 +140,18 @@ private TsPropertyModel getDecorators(SymbolTable symbolTable, TsBeanModel model
                        .collect(Collectors.toList());
     }
 
+    private boolean shouldNotBeDecorated(Class<?> type) {
+        return isBuiltIn(type) || isTsTypeResolutionUnsupported(type);
+    }
+
+    private boolean isTsTypeResolutionUnsupported(Class<?> type) {
+        return type.isEnum() || PresentPropertyJsonHierarchy.class.isAssignableFrom(type);
```

Comment:
Could you explain it to me why is resolution is supported if -> PresentPropertyJsonHierarchy.class.isAssignableFrom(type);

It is just question from my side to be able to better understand this feature :)

Reply:
This is just because resolution of hierarchies implemented using `PresentPropertyJsonHierarchy` has not been needed so far and is not implemented yet